src/provisioner/provisioner.py

```diff
@@ -168,15 +176,35 @@ def apply_cluster_config(cortx_conf, cortx_release):
 
             solution_config_nodes = CortxCluster(nodes, cortx_release)
             solution_config_nodes.save(cortx_conf)
-            solution_config_storagesets = CortxStorageSet(storage_sets)
-            solution_config_storagesets.save(cortx_conf)
+            if not CortxProvisioner._wait_for_lock_to_be_released(cortx_conf, CortxProvisioner._lock_timeout):
+                Conf.unlock(cortx_conf._conf_idx, lock_owner=Conf.machine_id, force = True, lock_key=CortxProvisioner._lock_key)
+                # TODO: remove Conf.save once gconf is completly moved to consul
+                Conf.save(cortx_conf._conf_idx)
+            if cortx_conf.get('cluster>storage_set') is None:
+                Conf.lock(cortx_conf._conf_idx, lock_owner=Conf.machine_id, lock_key=CortxProvisioner._lock_key)
+                # TODO: remove Conf.save once gconf is completly moved to consul
+                Conf.save(cortx_conf._conf_idx)
+                solution_config_storagesets = CortxStorageSet(storage_sets)
+                solution_config_storagesets.save(cortx_conf)
+                Conf.unlock(cortx_conf._conf_idx, lock_owner=Conf.machine_id, lock_key=CortxProvisioner._lock_key)
+                # TODO: remove Conf.save once gconf is completly moved to consul
+                Conf.save(cortx_conf._conf_idx)
         except KeyError as e:
             raise CortxProvisionerError(
                 errno.EINVAL,
                 f'Error occurred while applying cluster_config {e}')
 
     @staticmethod
-    def _get_node_info(_conf_idx: str):
+    def _wait_for_lock_to_be_released(cortx_conf: MappedConf, timeout: int):
```

Comment:
please have a default timeout or else place a condition to handle when timeout is None

Reply:
I am already passing default value from call. so I don't think it needs to be handled